src/coreclr/tools/aot/ILCompiler.ReadyToRun/Compiler/ReadyToRunStandaloneMethodMetadata.cs

```diff
@@ -0,0 +1,231 @@
+                    var exceptionRegion = exceptionRegions[i];
+
+                    _exceptionRegions[i] = new ILExceptionRegion(
+                        (ILExceptionRegionKind)exceptionRegion.Kind, // assumes that ILExceptionRegionKind and ExceptionRegionKind enums are in sync
+                        exceptionRegion.TryOffset,
+                        exceptionRegion.TryLength,
+                        exceptionRegion.HandlerOffset,
+                        exceptionRegion.HandlerLength,
+                        MetadataTokens.GetToken(exceptionRegion.CatchType),
+                        exceptionRegion.FilterOffset);
+                }
+            }
+
+            BlobReader localsBlob = default(BlobReader);
+            if (!_methodBody.LocalSignature.IsNil)
+            {
+                localsBlob = wrappedMethod.MetadataReader.GetBlobReader(wrappedMethod.MetadataReader.GetStandaloneSignature(_methodBody.LocalSignature).Signature);
+            }
+
+            AlternativeTypeRefProvider alternateTypes = new AlternativeTypeRefProvider();
+            EcmaSignatureEncoder<AlternativeTypeRefProvider> alternateEncoder = new EcmaSignatureEncoder<AlternativeTypeRefProvider>(alternateTypes);
+            Dictionary<int, int> _alternateNonTokenStrings = new Dictionary<int, int>();
+            BlobBuilder _alternateNonTypeRefStream = new BlobBuilder();
+            BlobBuilder _nonCodeAlternateBlob = new BlobBuilder();
+            {
+                // Generate alternate stream for exceptions.
+                _nonCodeAlternateBlob.WriteCompressedInteger(_exceptionRegions.Length);
+
+                for (int i = 0; i < _exceptionRegions.Length; i++)
+                {
+                    var region = _exceptionRegions[i];
+                    _nonCodeAlternateBlob.WriteCompressedInteger((int)region.Kind);
+                    _nonCodeAlternateBlob.WriteCompressedInteger((int)region.TryOffset);
+                    _nonCodeAlternateBlob.WriteCompressedInteger((int)region.TryLength);
+                    _nonCodeAlternateBlob.WriteCompressedInteger((int)region.HandlerOffset);
+                    _nonCodeAlternateBlob.WriteCompressedInteger((int)region.HandlerLength);
+                    if (region.Kind == ILExceptionRegionKind.Catch)
+                    {
+                        int alternateToken = GetAlternateStreamToken(region.ClassToken);
+                        int encodedToken = CodedIndex.TypeDefOrRefOrSpec(MetadataTokens.EntityHandle(alternateToken));
+                        _nonCodeAlternateBlob.WriteCompressedInteger(encodedToken);
+                    }
+                    else if (region.Kind == ILExceptionRegionKind.Filter)
+                    {
+                        _nonCodeAlternateBlob.WriteCompressedInteger(region.FilterOffset);
+                    }
+                }
+
+                if (localsBlob.Length == 0)
+                {
+                    // No locals. Encode a 0 to indicate this
```

Comment:
Could we just not write anything? I assume the length of the signature is part of the comparison at runtime, otherwise we would compare a method with no locals (0 written here) and a method with skiplocalsinit (0 written below) as equal.

Reply:
I want to be able to theoretically parse this thing in R2RDump one day. I'll change the encoding to encode a 2, which is reliably unique.